File: backend/apps/usuarios/routes/reporte_routes.py
from django.urls import path

from ..controllers.reporte_controller import (
    ResumenReservasView,
    ServiciosMasSolicitadosView,
    CitasCompletadasView,
    IngresosEstimadosView,
)

urlpatterns = [
    path("profesional/<int:profesional_id>/estadisticas/resumen/", ResumenReservasView.as_view(), name="resumen-reservas"),
    path("profesional/<int:profesional_id>/estadisticas/servicios/", ServiciosMasSolicitadosView.as_view(), name="servicios-mas-solicitados"),
    path("profesional/<int:profesional_id>/estadisticas/completadas/", CitasCompletadasView.as_view(), name="citas-completadas"),
    path("profesional/<int:profesional_id>/estadisticas/ingresos/", IngresosEstimadosView.as_view(), name="ingresos-estimados"),
]

File: backend/src/routes/reporte.routes.ts
import { Router } from "express";
import { ReporteController } from "../controllers/reporte.controller";
import { authenticateToken } from "../middlewares/auth.middleware";

const router = Router();
router.use(authenticateToken);
router.get("/profesional/:profesionalId/estadisticas/resumen/", ReporteController.resumen);
router.get("/profesional/:profesionalId/estadisticas/servicios/", ReporteController.servicios);
router.get("/profesional/:profesionalId/estadisticas/completadas/", ReporteController.completadas);
router.get("/profesional/:profesionalId/estadisticas/ingresos/", ReporteController.ingresos);
export default router;
